src/experiment.ts

```diff
@@ -0,0 +1,127 @@
+import { Identify } from 'amplitude-js';
+import { Client } from './client';
+
+const LOCAL_STORAGE_EXPERIMENTS_PREFIX = '__analytics_experiments_';
+
+/** Experiment interface allows getting a variation for a specified user. */
+export interface Experiment<Variation extends string> {
+	name: string;
+
+	engage(userId: string): Variation;
+}
+
+interface VariationData<Variation> {
+	variation: Variation;
+	fraction: number;
+	margin: number;
+}
+
+/** LocalExperiment allows describing an Experiment implemented with local storage. */
+export class LocalExperiment<Variation extends string>
+	implements Experiment<Variation> {
+	private readonly data: Array<VariationData<Variation>> = [];
+	private defaultVariation?: Variation;
+
+	constructor(
+		public readonly name: string,
+		private readonly analytics?: Client,
+	) {}
+
+	private static checkPercent(f: number) {
+		if (isNaN(f) || f < 0 || f > 100) {
+			throw new Error(
+				'Variation fraction must be defines as a percent value between 0 and 100',
+			);
+		}
+	}
+
+	private checkDuplicates(variation: Variation) {
+		const present = this.data.find(d => d.variation === variation);
```

Comment:
```suggestion
		const isPresent = this.data.some(d => d.variation === variation);
```

Reply:
The search result is then used to construct an error message, so I would stick with the current variant.